airflow/providers/snowflake/hooks/snowflake.py

```diff
@@ -305,15 +306,22 @@ def run(
             before executing the query.
         :param parameters: The parameters to render the SQL query with.
         :param handler: The result handler which is called with the result of each statement.
+        :param split_statements: Whether to split a single SQL string into statements and run separately
+        :param return_last: Whether to return result for only last statement or for all after split
+        :return: return only result of the LAST SQL expression if handler was provided.
         """
         self.query_ids = []
 
+        scalar_return_last = isinstance(sql, str) and return_last
         if isinstance(sql, str):
-            split_statements_tuple = split_statements(StringIO(sql))
-            sql = [sql_string for sql_string, _ in split_statements_tuple if sql_string]
+            if split_statements:
+                split_statements_tuple = util_text.split_statements(StringIO(sql))
+                sql = [sql_string for sql_string, _ in split_statements_tuple if sql_string]
+            else:
+                sql = [sql]
```

Comment:
i'm missing something.
Shouldn't this now be handled by the dbapi_hook?
Our goal was that providers will not need to split statments from their side but to rely on dbapi hook for it

Reply:
Correct. I missed it -ideally yes all the "split statement" code should be in the DBHool (now in common.sql). And we could even add >= 1.1  in case we add some features to common.sql . In case there are no breaking changes, this is perfectly fine to add >= 1.Y to any of the providers.

Reply:
All other hooks use `split_statements` from the DbApiHook except Snowflake. It has its own splitter and I'm not sure that `sqlparse.split` and `snowflake.connector.util_text.split_statements` behave similarly.
Also, they will provide new splitting conditions in a new version of the library, so it forced me to keep it in Snowflake hook

Reply:
If Snowflake provider decide to do things differently it's OK but I'm not sure there was a reason. I think it's just how the author of the code decided to do it.

I'm OK with leaving it as is with a TODO refrencing this discussion and asking to revisit it  in a followup PR + open a followup Issue on this task.

Reply:
I thought about the one additional parameter for this function, like:
```
def split_statements(sql, split_fn=None):
    if split_fn:
        return split_fn(sql)
    else:
        
```

But it will add unnecessary complexity. We could leave `snowflake.connector.util_text.split_statements` with TOD or use `DbApiHook.split_statements` and hope that nothing won't be broken.
Probably I'll try to find some Snowflake queries and provide an experiment